homeassistant/components/nobo_hub/sensor.py

```diff
@@ -0,0 +1,100 @@
+"""Python Control of Nobø Hub - Nobø Energy Control."""
+from __future__ import annotations
+
+from pynobo import nobo
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_IDENTIFIERS,
+    ATTR_MANUFACTURER,
+    ATTR_MODEL,
+    ATTR_NAME,
+    ATTR_SUGGESTED_AREA,
+    ATTR_VIA_DEVICE,
+    TEMP_CELSIUS,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from .const import ATTR_SERIAL, ATTR_ZONE_ID, DOMAIN, NOBO_MANUFACTURER
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up any temperature sensors connected to the Nobø Ecohub."""
+
+    # Setup connection with hub
+    hub: nobo = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_entities(
+        [
+            NoboTemperatureSensor(component["serial"], hub)
+            for component in filter(
+                lambda component: component[ATTR_MODEL].has_temp_sensor,
+                hub.components.values(),
+            )
+        ],
+        True,
+    )
+
+
+class NoboTemperatureSensor(SensorEntity):
+    """A Nobø device with a temperature sensor."""
+
+    _attr_device_class = SensorDeviceClass.TEMPERATURE
+    _attr_native_unit_of_measurement = TEMP_CELSIUS
+    _attr_state_class = SensorStateClass.MEASUREMENT
+    _attr_should_poll = False
+
+    def __init__(self, serial: str, hub: nobo) -> None:
+        """Initialize the temperature sensor."""
+        self._temperature: StateType = None
+        self._id = serial
+        self._nobo = hub
+        component = hub.components[self._id]
+        self._attr_unique_id = component[ATTR_SERIAL]
```

Comment:
If we may add more sensors in the future it's good to add a suffix here that represents the temperature measurement of the device. If that's not possible in the future, we don't need to do it.

Reply:
Temperature is the only sensor type that is supported by the hub.